[PM-21075] Add crypto abstraction layer for database seeder

rkac-bw opened the pull request:
## 🎟️ Tracking

<!-- Paste the link to the Jira or GitHub issue or otherwise describe / point to where this change is coming from. -->
https://bitwarden.atlassian.net/browse/PM-21075
## 📔 Objective

<!-- Describe what the purpose of this PR is, for example what bug you're fixing or new feature you're adding. -->
 This PR adds a crypto abstraction layer to the database seeder, enabling  Bitwarden encryption
  for seeded data and preparing for future Rust SDK integration.

  Key improvements:
  - Seeded users can now authenticate successfully 
  - Added vault item seeding capabilities (passwords, notes, cards, identities)
  - Possible future Rust SDK integration with feature flag support
  - Currently supports SQL Server and PostgreSQL (MySQL and SQLite to follow)

  ## 📝 Implementation Details
  - Introduced `ISeederCryptoService` interface to abstract crypto operations
  - Implemented C# version with proper Bitwarden encryption (PBKDF2, HKDF, AES-256-CBC)
  - Added `RustSeederCryptoService` stub showing one possible integration approach
  - Updated all recipes to use dependency injection
  - Added feature flag (`Seeder:UseRustSdk`) for future implementation switching
  - Commented out RustSdk file copy operations to prevent build errors (files don't exist yet)
  
## 📸 Screenshots

<!-- Required for any UI changes; delete if not applicable. Use fixed width images for better display. -->

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

Hinton committed:
SDK integration in db seeder

Commit:
[PM-21075] Add crypto abstraction layer for database seeder

- Introduce ISeederCryptoService interface for crypto operations
- Implement C# crypto service with proper Bitwarden encryption
- Add RustSeederCryptoService stub for future SDK integration
- Fix authentication by implementing correct double-hashing
- Add vault item seeding capabilities
- Update all recipes to use dependency injection
- Add feature flag for implementation switching
- Comment out RustSdk file copy operations to prevent build errors
- Currently supports SQL Server and PostgreSQL

Commit:
Apply dotnet format to fix CI build errors

Commit:
Add backward compatibility constructor for tests

withinfocus reviewed (changes requested):
I stopped about half-way, but there are solution-wide capabilities your AI generator wasn't aware of that replace a lot of this -- code in test and utility projects mainly to handle database configuration, data protection, and so on. This can thin out a lot and focus on the recipes utilizing the SDK. We'll need to get its invocation smoothed out too.

rkac-bw committed:
Fix whitespace formatting in OrganizationWithUsersRecipe

rkac-bw commented on bitwarden-server.sln:
```diff
@@ -133,6 +133,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Seeder", "util\Seeder\Seede
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "DbSeederUtility", "util\DbSeederUtility\DbSeederUtility.csproj", "{17A89266-260A-4A03-81AE-C0468C6EE06E}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RustSdk", "util\RustSdk\RustSdk.csproj", "{D1513D90-E4F5-44A9-9121-5E46E3E4A3F7}"
```
The RustSdk project was part of @Hinton's SDK integration work

rkac-bw commented on util/DbSeederUtility/ServiceCollectionExtension.cs:
```diff
@@ -8,18 +13,54 @@ namespace Bit.DbSeederUtility;
+        services.AddSingleton<IPasswordHasher<User>, PasswordHasher<User>>();
+
+        // Register ISeederCryptoService with feature flag support
+        // TODO: When Rust SDK is ready, check configuration to switch implementations
```
Agreed, just wanted to move closer to that point with this work

rkac-bw committed:
fix: general improvements and cleanup

- Re-add Skip attribute to performance test
- Clarify SDK integration TODO comment
- Clean up gitignore patterns
- Add project-specific gitignore for RustSdk
- Fix file formatting
